nixos/modules/services/misc/matrix-conduit.nix

```diff
@@ -0,0 +1,168 @@
+              description = "Max request size in bytes. Don't forget to also change it in the proxy.";
+            };
+            global.allow_registration = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether new users can register on this server.";
+            };
+            global.allow_encryption = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether new encrypted rooms can be created. Note: existing rooms will continue to work.";
+            };
+            global.allow_federation = mkOption {
+              type = types.bool;
+              default = false;
+              description = ''
+                Whether this server federates with other servers.
+                Upstream documentation contains the following warning:
+                You should not join Matrix rooms without asking the admins first.
+                We do not know whether Conduit is safe for general use yet, so you
+                should assume there is some chance that it breaks rooms permanently
+                for all participating users. We are not aware of such a bug today,
+                but we would like to do more testing.
+              '';
+            };
+            global.allow_jaeger = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether jaeger to support monitoring and troubleshooting through jaeger is enabled on this server.";
+            };
+            global.trusted_servers = mkOption {
+              type = types.listOf types.str;
+              default = [ "matrix.org" ];
+              description = "Servers trusted with signing server keys.";
+            };
+
+            global.address = mkOption {
+              type = types.str;
+              default = "127.0.0.1";
+              description = "Address to listen on for connections by the reverse proxy/tls terminator.";
+            };
+          };
+        };
+        default = {};
+        description = ''
+            generates the conduit.toml configuration file. Refer to
+            <link xlink:href="https://gitlab.com/famedly/conduit/-/blob/master/conduit-example.toml"/>
+            for details on supported values.
+            Note that database_path should not be set because the services reliance on systemd StateDir.
+        '';
+      };
+    };
+
+
+    config = mkIf cfg.enable {
+      services.conduit.settings = {
+        global.database_path = "/var/lib/matrix-conduit/conduit.db";
+      };
+      assertions = [
+        {
+          assertion = cfg.settings.global.server_name != null;
+          message = "services.conduit.settings.global.server_name must be set.";
+        }
+        {
+          assertion = cfg.settings.global.allow_federation -> cfg.iHaveReadTheFederationWarning;
+          message = ''
+            Upstream documentation contains the following warning:
```

Comment:
Maybe mention conduit somewhere here instead of just in the last line. Might be more helpful to the user to grasp where the message came from.